File: library/src/test/java/me/proxer/library/api/messenger/CreateConferenceEndpointTest.java
package me.proxer.library.api.messenger;

import me.proxer.library.ProxerTest;
import me.proxer.library.api.ProxerException;
import okhttp3.mockwebserver.MockResponse;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * @author Ruben Gees
 */
class CreateConferenceEndpointTest extends ProxerTest {

    @Test
    void testDefault() throws ProxerException, IOException {
        server.enqueue(new MockResponse().setBody(fromResource("create_conference.json")));

        final String result = api.messenger()
                .createConference("message", "someUser")
                .build()
                .execute();

        assertThat(result).isEqualTo("abcTest");
    }

    @Test
    void testPath() throws ProxerException, IOException, InterruptedException {
        server.enqueue(new MockResponse().setBody(fromResource("create_conference.json")));

        api.messenger().createConference("a", "b")
                .build()
                .execute();

        assertThat(server.takeRequest().getPath())
                .isEqualTo("/api/v1/messenger/newconference");
    }

    @Test
    void testParameters() throws ProxerException, IOException, InterruptedException {
        server.enqueue(new MockResponse().setBody(fromResource("create_conference.json")));

        api.messenger().createConference("test", "testUser")
                .build()
                .execute();

        assertThat(server.takeRequest().getBody().readUtf8())
                .isEqualTo("text=test&username=testUser");
    }
}

File: library/src/test/kotlin/me/proxer/library/api/messenger/CreateConferenceEndpointTest.kt
package me.proxer.library.api.messenger

import me.proxer.library.ProxerTest
import me.proxer.library.runRequest
import org.amshove.kluent.shouldBeEqualTo
import org.junit.jupiter.api.Test

/**
 * @author Ruben Gees
 */
class CreateConferenceEndpointTest : ProxerTest() {

    @Test
    fun testDefault() {
        val (result, _) = server.runRequest("create_conference.json") {
            api.messenger
                .createConference("message", "someUser")
                .build()
                .execute()
        }

        result shouldBeEqualTo "abcTest"
    }

    @Test
    fun testPath() {
        val (_, request) = server.runRequest("create_conference.json") {
            api.messenger.createConference("a", "b")
                .build()
                .execute()
        }

        request.path shouldBeEqualTo "/api/v1/messenger/newconference"
    }

    @Test
    fun testParameters() {
        val (_, request) = server.runRequest("create_conference.json") {
            api.messenger.createConference("test", "testUser")
                .build()
                .execute()
        }

        request.body.readUtf8() shouldBeEqualTo "text=test&username=testUser"
    }
}
